minerva/tools/calendar.py

```diff
@@ -6,73 +8,81 @@
-  """
-
-  # Remove the last three lines from the description if the third to last contains
-  # "Google Meet: https://"
-  lines = description.splitlines()
-  if len(lines) >= 3 and "Google Meet: https://" in lines[-3]:
-    return "\n".join(lines[:-3]).strip()
+    lines = description.splitlines()
+    if len(lines) >= 3 and "Google Meet: https://" in lines[-3]:
+        return "\n".join(lines[:-3]).strip()
 
 
 def parse_ics(ics_content: str) -> icalendar.Calendar:
-  return icalendar.Calendar.from_ical(ics_content)
+    return icalendar.Calendar.from_ical(ics_content)
 
 
 def query_calendar(cal: icalendar.Calendar, date_from: datetime, date_to: datetime) -> list[Event]:
-  filtered_cal = recurring_ical_events.of(cal).between(date_from, date_to)
-  events = []
-  for component in filtered_cal:
-    if component.name == "VEVENT":
-      summary = str(component.get("summary"))
-      description = _trim_google_meet_links_from_description(str(component.get("description", "")))
-      x_google_conference = str(component.get("x-google-conference"))
-      start = component.get("dtstart").dt
-      end = component.get("dtend").dt
-      rrule = component.get("rrule")
-      events.append(Event(summary, description, x_google_conference, start, end, rrule))
-  return events
+    filtered_cal = recurring_ical_events.of(cal).between(date_from, date_to)
+    events = []
+    for component in filtered_cal:
+        if component.name == "VEVENT":
+            summary = str(component.get("summary"))
+            description = _trim_google_meet_links_from_description(str(component.get("description", "")))
+            x_google_conference = str(component.get("x-google-conference"))
+            start = component.get("dtstart").dt
+            end = component.get("dtend").dt
+            rrule = component.get("rrule")
+            events.append(Event(summary, description, x_google_conference, start, end, rrule))
+    return events
 
 
 class CalendarTool:
-  def __init__(self, calendar_url: str):
-    calendar_request = httpx.get(calendar_url)
-    calendar_request.raise_for_status()
-    ics_content = calendar_request.text
-    self.cal = parse_ics(ics_content)
-
-  async def query(self, next_days: int) -> str:
-    """Query the event calendar for the next `next_days` days.
-
-    Args:
-      next_days: The number of days to query the calendar for. Should be at least 1. The max 366.
+    def __init__(self, calendar_url: str):
+        self.calendar_url = calendar_url
+        self.cal = None  # Initialize the calendar object as None
+        self.refetch_interval = int(os.getenv("CALENDAR_REFETCH_INTERVAL", 15))  # Fetch interval from env, default 15 minutes
+        asyncio.create_task(self._refetch_calendar_periodically())  # Start background task to refetch
+
+    async def _fetch_calendar(self):
+        """Fetch the calendar from the provided URL."""
+        try:
+            calendar_request = httpx.get(self.calendar_url)
```

Comment:
nit: you should either:
- use async `httpx` client to ensure that fetch the calendar asynchronously
- remove the `async` in front of `def` since we aren't doing anything async in this function

```suggestion
            async with httpx.AsyncClient() as client:
              calendar_request = await client.get(self.calendar_url)
```